tests/test_process_tv.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+"""Tests for medusa/process_tv.py."""
+import os
+
+from medusa.process_tv import ProcessResult
+
+import pytest
+
+
+@pytest.mark.parametrize('p', [
+    {
+        'path': 'media/postprocess/',
+        'nzb_name': None,
+        'failed': False,
+        'expected': True,
+        'structure': (
+            'bow.514.hdtv-lol[ettv].mkv',
+            'bow.514.hdtv-lol.srt',
+            {'samples': (
+                'sample.mkv', 'other.mkv',
+                {'inception': ()}
+            )}
+        )
+    },
+    {
+        'path': 'media/postprocess/',
+        'nzb_name': None,
+        'failed': False,
+        'expected': True,
+        'structure': (
+            'bow.514.hdtv-lol[ettv].mkv',
+            {'samples': (
+                'sample.mkv', 'other.mkv',
+                {'inception': (
+                    'cool.txt', 'bla.nfo'
+                )}
+            )}
+        )
+    },
+    {
+        'path': 'media/postprocess/',
+        'nzb_name': None,
+        'failed': False,
+        'expected': False,
+        'structure': (
+            'bow.514.hdtv-lol.srt',
+        )
+    },
+])
+def test_should_process(p, create_structure):
+    """Run the test."""
+    # Given
+    test_path = create_structure(p['path'], structure=p['structure'])
+    path = os.path.join(test_path, os.path.normcase(p['path']))
+    sut = ProcessResult(path)
+
+    # When
+    result = sut.should_process(path, p['nzb_name'], p['failed'])
```

Comment:
@medariox this should_process could have a different name to not confuse with should_process from post_processor. Maybe should_process_nzb